src/app/app.component.html

```diff
@@ -15,8 +15,36 @@ <h2>Partial language</h2>
             (buttonAction)="openFileSelector()"
             (dropAction)="dropFiles($event)"
         ></app-template-button>
-        <mat-form-field appearance="outline" class="textarea">
-            <mat-label>Source file</mat-label>
+        <mat-form-field
+            appearance="outline"
+            class="textarea"
+            [matTooltip]="runHint"
+        >
+            <mat-label
+                class="source_label"
+                fxLayout="row"
+                fxLayoutAlign="space-between"
+            >
+                <span>Source file</span>
+                <mat-icon
+                    *ngIf="isRunValid === Valid.Success"
+                    class="color_success"
+                >
+                    check_circle
+                </mat-icon>
+                <mat-icon
+                    *ngIf="isRunValid === Valid.Warn"
+                    class="color_warning"
+                >
+                    warning_amber
+                </mat-icon>
+                <mat-icon
+                    *ngIf="isRunValid === Valid.Error"
+                    class="color_error"
+                >
+                    error_outline
+                </mat-icon>
```

Comment:
_Nice to have:_

Hier könnte man auch gut mit einem Switch arbeiten:

```suggestion
                <ng-container [ngSwitch]="runValidationStatus">
                    <mat-icon *ngSwitchCase="'success'" class="color_success">
                        check_circle
                    </mat-icon>
                    <mat-icon *ngSwitchCase="'warn'" class="color_warning">
                        warning_amber
                    </mat-icon>
                    <mat-icon *ngSwitchCase="'error'" class="color_error">
                        error_outline
                    </mat-icon>
                </ng-container>
```